app/models/pipeline_run.rb

```diff
@@ -319,25 +320,32 @@ def db_load_ercc_counts
     update(total_ercc_reads: total_ercc_reads)
   end
 
-  def db_load_contigs
+  def db_load_contigs(contig2taxid)
     contig_stats_s3_path = s3_file_for("contigs")
     contig_s3_path = "#{postprocess_output_s3_path}/#{ASSEMBLED_CONTIGS_NAME}"
 
     downloaded_contig_stats = PipelineRun.download_file_with_retries(contig_stats_s3_path,
                                                                      LOCAL_JSON_PATH, 3)
     contig_stats_json = JSON.parse(File.read(downloaded_contig_stats))
     return if contig_stats_json.empty?
+
     update(assembled: 1)
     contig_fasta = PipelineRun.download_file_with_retries(contig_s3_path, LOCAL_JSON_PATH, 3)
     contig_array = []
+    taxid_list = []
+    contig2taxid.values.each { |entry| taxid_list += entry.values }
+    taxon_lineage_map = {}
+    TaxonLineage.where(taxid: taxid_list.uniq).each { |t| taxon_lineage_map[t.taxid.to_i] = t.to_a }
```

Comment:
do you need to select only the most recent / the most appropriate?

Reply:
great point. i can make it oder by id so the latest one will always be there.